### PR TITLE
Strategy from paper

### PR DESCRIPTION
**Pull request by rank-and-file**
Beginning the implementation of the strategy from the paper

### CODE REVIEW

File: src/main/java/com/treasure/hunt/strategy/searcher/implementations/StrategyFromPaper.java [hunk 1]
```diff
@@ -21,16 +38,119 @@ public void commitProduct(Product product) {
...
+        double startX = start.getX();
+        double startY = start.getY();
+        A = JTSUtils.createPoint(startX-halfDiff, startY+halfDiff);
+        B = JTSUtils.createPoint(startX+halfDiff, startY+halfDiff);
+        C = JTSUtils.createPoint(startX-halfDiff, startY+halfDiff);
+        D = JTSUtils.createPoint(startX-halfDiff, startY-halfDiff);
+    }
+
+
+    private Moves RectangeScan(Point A, Point B, Point C, Point D){
```
Comment (jotoh98) [inline src/main/java/com/treasure/hunt/strategy/searcher/implementations/StrategyFromPaper.java]:
```suggestion
    private Moves rectangleScan(Point A, Point B, Point C, Point D){
```

File: src/main/java/com/treasure/hunt/utils/JTSUtils.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            if(a2/c2 == a1/c1){
+                return new Point[]{P1, P2, Q1, Q2}; // the lines are equal
+            }
+            return new Point[0];
+        }
+        double x = (b2*c1 - b1*c2)/determinant;
+        double y = (a1*c2 - a2*c1)/determinant;
+        return new Point[]{createPoint(x, y)};
+    }
```
Comment (jotoh98) [inline src/main/java/com/treasure/hunt/utils/JTSUtils.java]: please see org.locationtech.jts.geom. LineSegment.intersection(). 
Maybe this function already exists.

File: src/main/java/com/treasure/hunt/utils/JTSUtils.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        if(intersection.length==0)
+            return intersection;
+
+        if((intersection[0].intersection(createLineString(S1,S2))).isEmpty())
+            return new Point[0];
+
+        return intersection;
+    }
+}
```
Comment (jotoh98) [inline src/main/java/com/treasure/hunt/utils/JTSUtils.java]: Wrapper for lineIntersection()? Can this also be replaced by the JTS function mentioned above?

File: src/main/java/com/treasure/hunt/strategy/hint/HalfplaneHint.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                            // right and left indicate where the target is (right indicates the target is in positive x-Direction
+                            // in relationship to the line)
+                            // when the line is horizontal, up signals the target is in positive y-Direction in relationship
+                            // to the line (the up and down enumerators are only used when the line is horizontal)
+                            // left and down respectively
+
+    public static HalfplaneHint angular2correctHalfPlaneHint(AngleHint anglehint)
+    {
+
+
```
Comment (jotoh98) [inline src/main/java/com/treasure/hunt/strategy/hint/HalfplaneHint.java:32]: Plase pull latest master, we have committed some code styling preferences to enhance readabilty. Also, search for shortcuts reformatting your code. More than 1 blank line inside of code declaration is disabled by default.

**Review by jotoh98 — CHANGES_REQUESTED** [review]
Overall good work, some implementations seem missing or in work in the algorithm implementation. However, there are some real issues with the DRY principle, please don't implement functionalities twice which are already available with JTS.

File: src/main/java/com/treasure/hunt/strategy/searcher/movesImplementations/Steps.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.treasure.hunt.strategy.searcher.movesImplementations;
+
+import com.treasure.hunt.strategy.searcher.Moves;
+import org.locationtech.jts.geom.Point;
+
+public class Steps extends Moves {
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/strategy/searcher/movesImplementations/Steps.java]: Why ?

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/treasure/hunt/strategy/searcher/movesImplementations/Steps.java (hunk 5 above)
Comment (1-alex98) [reply src/main/java/com/treasure/hunt/strategy/searcher/movesImplementations/Steps.java]: just use Moves class

**Comment by 1-alex98** [comment]
Could u merge master in so u stay up to date

**Pull request closed without merging**